Windows path fixes

ranaur opened the pull request:
I hope this time is right.

In the spirit of Halloween, I'll cite the witch from Woodpecker "Witch Crafty" catoon: "in the way we go!" :--)

MichaelAquilina committed:
Correct package metadata in setup.py

MichaelAquilina committed:
Autoplay asciinema

MichaelAquilina committed:
Bump version to 0.1.30

MichaelAquilina committed:
Correct License entry

MichaelAquilina committed:
Bump version to 0.1.31

MichaelAquilina committed:
Add basic working prototype of command to watch targets

MichaelAquilina committed:
Add read_delay and only run once per target hit

MichaelAquilina committed:
Revert change to sync_command

MichaelAquilina committed:
Add basic coverage test for daemon command

MichaelAquilina committed:
Add enum34 to requirements.txt

MichaelAquilina committed:
Add --read-delay parameter

MichaelAquilina committed:
Minor correction to README

MichaelAquilina committed:
Add basic support for syncing specific keys

MichaelAquilina committed:
Add conflicts option override

MichaelAquilina committed:
Add recursive INotify support

MichaelAquilina committed:
Simplify pip cache declaration on travis

MichaelAquilina committed:
Numerous fixes and improvements

MichaelAquilina committed:
Add support for displaying timestamps in log messages

MichaelAquilina committed:
Remove keys from command line for sync command

MichaelAquilina committed:
Add tests for loglevel and timestamps

MichaelAquilina committed:
Update help for daemon command

MichaelAquilina committed:
Add tests for INotifyRecursive

MichaelAquilina committed:
Add a timeout to INotifyRecursive tests

MichaelAquilina committed:
Fix flake8 linter issues

MichaelAquilina committed:
Setup flake8-import-order

MichaelAquilina committed:
Add tests for daemon command

MichaelAquilina committed:
Add entry in README about daemon command

MichaelAquilina committed:
Make ignore_files and index lazy on S3SyncClient and add repr test for
SyncWorker

MichaelAquilina committed:
Merge pull request #63 from MichaelAquilina/inotify_daemon

Add basic working prototype of command to watch targets

MichaelAquilina committed:
Bump version to 0.2.0

MichaelAquilina committed:
Bump to version 0.2.1

MichaelAquilina committed:
Add build folder to .gitignore

MichaelAquilina committed:
Use "pip install ." instead of using setup.py directly

MichaelAquilina committed:
Add filelock to requirements

MichaelAquilina committed:
Implement basic file locking on the LocalSyncClient

MichaelAquilina committed:
Test local locking mechanism

MichaelAquilina committed:
Lock client targets when running sync

MichaelAquilina committed:
Merge pull request #68 from MichaelAquilina/target_locking

Target locking

MichaelAquilina committed:
Bump version to 0.2.2

MichaelAquilina committed:
Expand user paths when editing targets

MichaelAquilina committed:
Bump version to 0.2.3

MichaelAquilina committed:
Show Stack Trace for exceptions if log level is DEBUG

MichaelAquilina committed:
Fix failing test and add new test for showing stack trace

MichaelAquilina committed:
Create and remove target lockfile

MichaelAquilina committed:
Bump version to 0.2.4

MichaelAquilina committed:
Print rather than log ls output so that it goes to stdout

MichaelAquilina committed:
Update ls_command tests

MichaelAquilina committed:
Merge pull request #74 from MichaelAquilina/ls_command_stdout

Print ls command stdout

MichaelAquilina committed:
Fix minor error in README

MichaelAquilina committed:
Use capsys fixture in tests

MichaelAquilina committed:
Merge pull request #76 from MichaelAquilina/capsys_fixture

Use capsys fixture in tests

MichaelAquilina committed:
Use stdout for printing targets output

MichaelAquilina committed:
Correct use of print

MichaelAquilina committed:
Reintroduce flake8 into CI

MichaelAquilina committed:
Explicitly specific folders to exclude from flake8 linting

MichaelAquilina committed:
Remove usage of DeferredFunction and move CLI code out of sync.py

MichaelAquilina committed:
Add dry-run option for sync command

MichaelAquilina committed:
TestResolution

MichaelAquilina committed:
Cleanup and fix all failing tests

MichaelAquilina committed:
More tests for Resolution class

MichaelAquilina committed:
Add conflict handler to cli client

MichaelAquilina committed:
Bump alpha version

MichaelAquilina committed:
Singleton wrapper implementation of progressbar

MichaelAquilina committed:
Use start, update and complete callback

MichaelAquilina committed:
Make string layout more clear for handle_conflict

MichaelAquilina committed:
Move conflict_handler to constructor

MichaelAquilina committed:
Add test for dry runs

MichaelAquilina committed:
Use constants for actions

MichaelAquilina committed:
Simplify conditional

MichaelAquilina committed:
Add test for callback

MichaelAquilina committed:
Add test_put_callback for s3 client

MichaelAquilina committed:
Move Resolution class to its own file

MichaelAquilina committed:
test_get_resolution

MichaelAquilina committed:
Test Handle Conflict

MichaelAquilina committed:
Add smoke test for progress bar callbacks

MichaelAquilina committed:
Merge pull request #78 from MichaelAquilina/isolate_cli_functionality

Isolate cli functionality

MichaelAquilina committed:
Remove colored from sync.py

MichaelAquilina committed:
test_action_callback

MichaelAquilina committed:
Merge pull request #81 from MichaelAquilina/remove_colored_from_sync

Remove colored from sync.py

epayet committed:
Add --no-colors option

epayet committed:
Add Contributing section to readme

epayet committed:
Move libmagic installation to Installation section

MichaelAquilina committed:
Merge pull request #83 from epayet/readme-contributing

Add Contributing section to readme

MichaelAquilina committed:
Add --copy-target-credentials

MichaelAquilina committed:
Dont crash on most commands if config doesnt exist

MichaelAquilina committed:
Merge pull request #87 from MichaelAquilina/prebuilt_config

Dont crash on most commands if config doesnt exist

MichaelAquilina committed:
Merge pull request #88 from MichaelAquilina/copy_settings

Add --copy-target-credentials

MichaelAquilina committed:
Move INotifyRecursive to its own file

MichaelAquilina committed:
Move ProgressBar to its own file

MichaelAquilina committed:
Remove redundant code

MichaelAquilina committed:
No need for ident on config dump

MichaelAquilina committed:
Move show_diff to its own file

MichaelAquilina committed:
Merge pull request #89 from MichaelAquilina/cleanup

Code clean up and refactoring

MichaelAquilina committed:
Only catch KeyboardInterrupts in the cli client

MichaelAquilina committed:
Merge pull request #90 from MichaelAquilina/keyboard_interrupt_cli

Only catch KeyboardInterrupts in the cli client

MichaelAquilina committed:
Ignore changes to .s4lock when running the daemon command

MichaelAquilina committed:
Merge pull request #92 from MichaelAquilina/fix_daemon

Ignore changes to .s4lock when running the daemon command

MichaelAquilina committed:
Merge pull request #82 from epayet/no-colors

Add --no-colors option

MichaelAquilina committed:
Move sync command to SyncCommand

MichaelAquilina committed:
Move ls command to LsCommand

MichaelAquilina committed:
Move targets command to TargetsCommand

MichaelAquilina committed:
Move add command to AddCommand

MichaelAquilina committed:
Use fixture rather than yield_fixture

MichaelAquilina committed:
Fix ciruclar import

MichaelAquilina committed:
Move daemon command to DaemonCommand class

MichaelAquilina committed:
Move edit command to EditCommand class

MichaelAquilina committed:
Move rm command to RmCommand class

MichaelAquilina committed:
Commands base class

MichaelAquilina committed:
Merge pull request #94 from MichaelAquilina/commands_as_classes

Commands as classes

MichaelAquilina committed:
Re-add missing entry point

MichaelAquilina committed:
Check that version output is non empty

MichaelAquilina committed:
Merge pull request #95 from MichaelAquilina/fix_missing_entry

Re-add missing entry point to cli

MichaelAquilina committed:
pathlib backport now required for python 3.3 support

MichaelAquilina committed:
Merge pull request #98 from MichaelAquilina/fix_python_33

pathlib backport now required for python 3.3 support

MichaelAquilina committed:
Remove pathlib now that its been fixed in inotify_simple

MichaelAquilina committed:
Merge pull request #99 from MichaelAquilina/remove_pathlib

Remove pathlib now that its been fixed in inotify_simple

MichaelAquilina committed:
Only display stack traces if log-level=DEBUG

MichaelAquilina committed:
Merge pull request #101 from MichaelAquilina/better_failures

Only display stack traces if log-level=DEBUG

MichaelAquilina committed:
Bump version to 0.2.6

MichaelAquilina committed:
Fix failure when .s4lock is not found

MichaelAquilina committed:
Upgrade s4 to 0.2.7

MichaelAquilina committed:
Remove 3.7-dev from travis builds

MichaelAquilina committed:
Dont crash on OSX because INotify is unsupported

MichaelAquilina committed:
Add note in README about Daemon command support

MichaelAquilina committed:
Add comment about import catch in Daemon command

MichaelAquilina committed:
Merge pull request #105 from MichaelAquilina/fix_macosx

Dont crash on OSX because INotify is unsupported

MichaelAquilina committed:
Upgrade s4 to 0.2.8

MichaelAquilina committed:
Minor spelling fix

MichaelAquilina committed:
Add mypy_cache to gitignore

MichaelAquilina committed:
Add information about configuration file location

MichaelAquilina committed:
Correct rst formatting

MichaelAquilina committed:
Quote array expansion to prevent word splitting

MichaelAquilina committed:
Add mypy and pytest cache to gitignore

MichaelAquilina committed:
Upgrade version to 0.2.9

MichaelAquilina committed:
Upgrade dev-requirements to the latest versions

MichaelAquilina committed:
Drop support for python 3.3

MichaelAquilina committed:
Add CHANGELOG.rst

MichaelAquilina committed:
Remove python 3.3 from travis builds

MichaelAquilina committed:
Merge pull request #107 from MichaelAquilina/0.2.9

0.2.9

MichaelAquilina committed:
Add Sections to the beginning of the README

MichaelAquilina committed:
Update CHANGELOG

MichaelAquilina committed:
Fix "Other Subcommands" title

MichaelAquilina committed:
Merge pull request #109 from MichaelAquilina/sections_readme

0.2.10

MichaelAquilina committed:
Bump version to 0.2.10

MichaelAquilina committed:
Fix rST on pypi

MichaelAquilina committed:
Use gzip compression by default for s3 indexes

MichaelAquilina committed:
Minor linter fix

MichaelAquilina committed:
Return decompressed and decoded index

MichaelAquilina committed:
Bump version to 0.2.12

MichaelAquilina committed:
Correct content type for gzip

MichaelAquilina committed:
Test loading different compression formats in s3 index

MichaelAquilina committed:
Assume octect-streams are zlib compressed indexes

MichaelAquilina committed:
Merge pull request #112 from MichaelAquilina/use_gzip_s3

Use gzip compression by default for s3 indexes

MichaelAquilina committed:
Remove unused py37 entry from tox.ini

MichaelAquilina committed:
Add command short aliases

MichaelAquilina committed:
Update CHANGELOG and bump version

MichaelAquilina committed:
Merge pull request #115 from MichaelAquilina/command_short_aliases

Add command short aliases

MichaelAquilina committed:
Fix minor spelling error on CHANGELOG

MichaelAquilina committed:
Add parameterized tests for command aliases

MichaelAquilina committed:
Update CHANGELOG and bump version

MichaelAquilina committed:
Remove bin/s4

MichaelAquilina committed:
Add entry_point to setup.py

MichaelAquilina committed:
Several improvements to setup.py

MichaelAquilina committed:
Fix minor flake8 issue

MichaelAquilina committed:
Merge pull request #118 from MichaelAquilina/console_scripts

Use console_scripts to generate executable shim

MichaelAquilina committed:
Bump version to 0.2.15

MichaelAquilina committed:
Add ISSUE_TEMPLATE

MichaelAquilina committed:
Merge pull request #120 from MichaelAquilina/add_issue_template

Add ISSUE_TEMPLATE

Commit:
Replace fnmatch with pathspac for more thorough gitignore support

Commit:
Reference item.name instead of item in path join

MichaelAquilina committed:
Merge pull request #122 from troylar/replace_fnmatch_with_pathspec

Replace fnmatch with pathspec for more thorough gitignore support

MichaelAquilina committed:
Bump version to 0.2.16

MichaelAquilina committed:
Update CHANGELOG

MichaelAquilina committed:
Match s4 version with git version and ensure entry in CHANGELOG

MichaelAquilina committed:
Merge pull request #126 from MichaelAquilina/version_linting

Version linting

MichaelAquilina committed:
Format all code using black

MichaelAquilina committed:
Merge pull request #128 from MichaelAquilina/black_formatted

Format all code using black

MichaelAquilina committed:
Add Snyk badge to README

MichaelAquilina committed:
Remove read the docs badge

MichaelAquilina committed:
Merge pull request #129 from MichaelAquilina/snyk_badge

Add Snyk badge to README

MichaelAquilina committed:
Add missing main

MichaelAquilina committed:
Add docstrings explaining intended functionality of SyncClient methods

MichaelAquilina committed:
Add get_size method to LocalSyncClient

MichaelAquilina committed:
Calculate total size with ls output

MichaelAquilina committed:
Merge pull request #131 from MichaelAquilina/total_size

Total size

MichaelAquilina committed:
Bump version to 0.2.18

MichaelAquilina committed:
Migrate to CircleCI 2.0 for builds

MichaelAquilina committed:
Remove extra dependencies from dev-requirements.txt

MichaelAquilina committed:
Merge pull request #136 from MichaelAquilina/circleci

Migrate to CircleCI 2.0

MichaelAquilina committed:
Remove scandir and enum34 as dependencies

MichaelAquilina committed:
Merge pull request #141 from MichaelAquilina/reduce_dependencies

Remove scandir and enum34 as dependencies

MichaelAquilina committed:
Add monthly scheduled test

MichaelAquilina committed:
Merge pull request #142 from MichaelAquilina/scheduled_builds

Add monthly scheduled test

MichaelAquilina committed:
Add pyproject.toml

MichaelAquilina committed:
Add poetry dependencies

MichaelAquilina committed:
Use poetry in CircleCI

MichaelAquilina committed:
run pytest in poetry

MichaelAquilina committed:
Remove setup.py

MichaelAquilina committed:
Install poetry for lint job

MichaelAquilina committed:
Add repository section to pyproject.toml

MichaelAquilina committed:
Add homepage and keywords entry to pyproject.toml

MichaelAquilina committed:
Fix python requires

MichaelAquilina committed:
Fix description

MichaelAquilina committed:
Update dependencies

MichaelAquilina committed:
Add scripts entry

MichaelAquilina committed:
Remove Snyk

MichaelAquilina committed:
Use poetry run to test s4 script

MichaelAquilina committed:
Better visibility of version lint process

MichaelAquilina committed:
poetry install

MichaelAquilina committed:
poetry install --no-dev

MichaelAquilina committed:
lint pyproject version

MichaelAquilina committed:
Remove outdated Running locally section

MichaelAquilina committed:
Merge pull request #143 from MichaelAquilina/poetry

Poetry

MichaelAquilina committed:
Bump version to 0.2.19

MichaelAquilina committed:
Add missing readme entry to pyproject.toml

MichaelAquilina committed:
Merge pull request #145 from MichaelAquilina/fix_missing_readme

Add missing readme entry to pyproject.toml

MichaelAquilina committed:
Bump version to 0.2.20

MichaelAquilina committed:
Remove unused logger fixture

MichaelAquilina committed:
Fix lint errors

MichaelAquilina committed:
Fix mistakes in test_sync

MichaelAquilina committed:
Update dependencies

MichaelAquilina committed:
Merge pull request #146 from MichaelAquilina/test_fixes

Various Updates

MichaelAquilina committed:
Remove unused closure

MichaelAquilina committed:
Introduce isort

MichaelAquilina committed:
Merge pull request #147 from MichaelAquilina/more_cleanup

Introduce isort

MichaelAquilina committed:
Remove impossible to reach code

MichaelAquilina committed:
Merge pull request #148 from MichaelAquilina/increase_codecov

Remove impossible to reach code

MichaelAquilina committed:
Add black badge to README

MichaelAquilina committed:
Merge pull request #149 from MichaelAquilina/black_badge

Add black badge to README

MichaelAquilina committed:
Minor changes to README

MichaelAquilina committed:
Merge pull request #150 from MichaelAquilina/readme_update

Minor changes to README

MichaelAquilina committed:
Fix broken poetry build

MichaelAquilina committed:
Merge pull request #158 from MichaelAquilina/fix_poetry

Fix broken poetry build

MichaelAquilina committed:
Dont support daemon command on Windows

MichaelAquilina committed:
Merge pull request #161 from MichaelAquilina/daemon_command_no_support_windows

Dont support daemon command on Windows

MichaelAquilina committed:
Update to version 0.2.21

MichaelAquilina committed:
Add support for other services like Minio

MichaelAquilina committed:
Merge pull request #162 from MichaelAquilina/support_minio

Add support for other services like Minio

MichaelAquilina committed:
Minor release 0.3.0

MichaelAquilina committed:
Add Running from Source section to README

MichaelAquilina committed:
Merge pull request #163 from MichaelAquilina/running_from_source

Add Running from Source section to README

MichaelAquilina committed:
Improve behaviour of the add command

MichaelAquilina committed:
Numerous improvements to test involving get_input

MichaelAquilina committed:
Merge pull request #164 from MichaelAquilina/better_add_command

Improve behaviour of the add command

MichaelAquilina committed:
Upgrade to version 0.4.0

MichaelAquilina committed:
Update poetry lock file

Commit:
Reset to master branch

Merge branch 'master' of https://github.com/MichaelAquilina/S4

Commit:
Fixed handling of path separator, allowing at the sema time using "\" in windows
and "/" in unix machines.

MichaelAquilina commented:
Ok so there are still clearly some issue with the way this PR is based off master. But I think I can help you figure it out.

Run the following commands:

```
git checkout stash
git checkout master
git pull upstream master
git checkout windows_path_fixes
git reset --hard master
git cherry-pick ab15a66
git push --force origin HEAD
```

After you run those commands, we should be at a good stage to start reviewing the code!

MichaelAquilina commented:
Something is strange about this PR :thinking: It's still showing as wanting to push 280+ commits (despite you only wanting to push 1). When you ran the commands, did you get any weird messages?

I'll review the code itself this evening :)

ranaur commented:
No. Only the first command was not "git checkout stash", but "git stash".
But the rest was simple.

Here is the git log. Note there are only two commits from me.

After the
commit 62ceed99b653047fb2754c15890952db2f8d9bf8 Author: ranaur <
ranaur2008@ranaur.net> Date: Thu Nov 1 12:23:04 2018 -0300 Fixed handling
of path separator, allowing at the sema time using "\" in win dows and "/"
in unix machines. commit ede2d6c063dcc1721fc60708e034a769f1328fc1 Merge:
0dd2b1f cc26a45 Author: ranaur <ranaur2008@ranaur.net> Date: Thu Nov 1
12:12:27 2018 -0300 Reset to master branch Merge branch 'master' of
https://github.com/MichaelAquilina/S4 *commit
0dd2b1f6fc882eb1ada2152907460f9713313a84 *Author: Michael Aquilina <
michaelaquilina@gmail.com> Date: Tue Oct 30 23:23:08 2018 +0000 Update
poetry lock file
...

On Mon, Nov 5, 2018 at 12:04 PM Michael Aquilina <notifications@github.com>
wrote:

> Something is strange about this PR 🤔 It's still showing as wanting to
> push 280+ commits (despite you only wanting to push 1). When you ran the
> commands, did you get any weird messages?
>
> I'll review the code itself this evening :)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/MichaelAquilina/S4/pull/167#issuecomment-435886447>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACANM7blbfq43Z5WQ5G-2m5jC6jN1Tvhks5usEWIgaJpZM4YHGVr>
> .
>

MichaelAquilina commented:
> No. Only the first command was not "git checkout stash", but "git stash"
:man_facepalming: yeah sorry that was my mistake. Nice catch though!

Commit:
Add watcher library

Commit:
Added support of the daemon command for Windows (may break Linux compatibility)

ranaur commented:
I just sent another commit. This time I added support for the daemon
command on windows.

It wasn't easy. There is no INotify on windows, so I wrote a base class
(quite similar to INotifyRecursive, but it is not the same). it is called
watchdir. There is two implementations of the class depending on the
operating system. I haven't tested the Linux version yet, I'll do it at
home.

I also added a test file for this class (WatchDir) on test directory. I
didn't mock anything. Actually the test is quite complex. It simulates the
creation of diles and directories and try to catch if the class is really
watching for them.


On Mon, Nov 5, 2018 at 1:04 PM Michael Aquilina <notifications@github.com>
wrote:

> No. Only the first command was not "git checkout stash", but "git stash"
> 🤦‍♂️ yeah sorry that was my mistake. Nice catch though!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/MichaelAquilina/S4/pull/167#issuecomment-435907031>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACANMx1eH3iXAVvXBxBzM_odAOIIL8XLks5usFNpgaJpZM4YHGVr>
> .
>

MichaelAquilina commented:
Thanks @ranaur! The code looks quite complex. So I would suggest to keep it simple we have a separate Pull Request for just the daemon code.

Looking at the path fixes, I might have found a better alternative to what you are proposing which I'll try out when I get the chance. 

Your git history still seems to be problematic I'm afraid :/ go directly to github and click on "commits". There should only be yours but you seem to have the entire history of master.

I would strongly advise you interact with github.com directly rather than using email as it will make it easier to see what I mean (It also means that if I happen to update a comment you see the most up to date version - which you wont get on email!)
